components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/decoder/ThumbnailDecoder.kt

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.thumbnails.decoder
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.util.Size
+import androidx.annotation.VisibleForTesting
+import mozilla.components.support.base.log.logger.Logger
+import kotlin.math.floor
+import kotlin.math.max
+import kotlin.math.min
+
+/**
+ * A thumbnail decoder that can decode a [ByteArray] into a [Bitmap].
+ */
+class ThumbnailDecoder {
```

Comment:
Since this is very close to the icon decoder implementation: Do you think there's enough overlap so that we could extract it and share it via a `support-*` component?

Reply:
Fixed in https://github.com/mozilla-mobile/android-components/pull/7003